File: domain/src/main/java/com/chronos/mobile/domain/TimetableRepository.kt
package com.chronos.mobile.domain

import com.chronos.mobile.core.model.AppState
import com.chronos.mobile.core.model.Course
import com.chronos.mobile.core.model.Timetable
import kotlinx.coroutines.flow.Flow

interface TimetableRepository {
    val appState: Flow<AppState>

    suspend fun getAppStateSnapshot(): AppState
    suspend fun getTimetable(id: String): Timetable?
    suspend fun saveTimetable(timetable: Timetable)
    suspend fun saveCourse(timetableId: String, course: Course)
    suspend fun deleteCourse(courseId: String)
    suspend fun deleteTimetable(id: String)
    suspend fun setCurrentTimetableId(id: String?)
    suspend fun setWallpaper(uri: String?)
    suspend fun decodeTimetable(json: String): Timetable
    suspend fun encodeTimetable(timetable: Timetable): String
}

File: src/lib/domain/interfaces/timetable-repository.ts
import type { AppState } from '$lib/models/app-state';
import type { Course } from '@chronos/core';
import type { Timetable } from '$lib/models/timetable';

export interface TimetableRepository {
	subscribeAppState(listener: (state: AppState) => void): () => void;
	getAppStateSnapshot(): Promise<AppState>;
	getTimetable(id: string): Promise<Timetable | null>;
	saveTimetable(timetable: Timetable): Promise<void>;
	saveCourse(timetableId: string, course: Course): Promise<void>;
	deleteCourse(courseId: string): Promise<void>;
	deleteTimetable(id: string): Promise<void>;
}
